app/src/main/java/com/example/weather/network/Dao.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.weather.network
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+
+@Dao
+interface CityOverviewDao {
+
+    @Query("SELECT * FROM city_overview_table where name = name")
+    fun getCities(): List<CachedCityOverview>
+
+    @Insert
+    suspend fun insert(cachedCity: CachedCityOverview)
+
+    @Query("DELETE FROM city_overview_table")
+    suspend fun deleteAll()
+}
+
+@Dao
+interface CityDetailsDao {
+
+    @Query("SELECT * FROM city_details_table ORDER BY name ASC")
+    fun getCityData(): List<CachedCityDetails>
```

Comment:
this should take in a city id parameter so you know what city to fetch details data for. you'll then want to use that parameter in the SQL query to fetch the right data